### PR TITLE
Dev Restructure Proposal

### PR DESCRIPTION
**Pull request by talves**
**- Summary**

Develop using `example/site/admin` structure in development.

**- Test plan**

Run through all development tests with a new clone.
Run `yarn` , `yarn start` to make sure all is compiling well.
Run local server tests.

**- Description for the changelog**

First step in creating a cohesive solution for running a cli and development.
Cleans up the development builds and loaders to stop asset errors.

### CODE REVIEW

**Comment by talves** [comment]
@erquhart mentioned a root deploy. This should be simple with a deploy of the `/example/site/admin` path after the copy.

**Comment by talves** [comment]
The branch config.yml in #554 is setup to point to a different branch for backend. This is due to keeping clutter from the PR branch. For consistency, backend branches created for the sole purpose of testing should be created as a branch of the PR submitted for merge.

I feel we should consider this in this Dev Restructure proposal.
I fixed all the asset path errors with this PR, It should have a check for NODE_ENV=development to change asset paths to a deploy preview site path for testing ***WHEN** there is a deploy-site-path in the `config.yml` or maybe in process.env.TESTING_SITE

I will test the configuration and see how easy it would be.

Thoughts?

**Comment by erquhart** [comment]
@talves good thoughts here. Cleaning up errors is awesome, thank you for that! I also totally agree on using a branch off of the PR branch for testing backends, that's a much better approach.

**Comment by Benaiah** [comment]
@talves I like this restructure, and I think your idea about PR backend branches is a good one.

**Comment by talves** [comment]
@Benaiah and @erquhart I am glad you guys agree on the PR backend branches.

Question; ***Can we test a backend branch with a folder path in the CMS?***
In example `test/fix-pr-000`

I also feel we need to folder path test branches, so we can easily identify and clean them out of the repository and they do not clutter. Just a thought.

File: webpack.dev.js [hunk 1]
```diff
@@ -48,17 +48,14 @@ module.exports = merge.smart(require('./webpack.base.js'), {
     }),
     new webpack.HotModuleReplacementPlugin(),
     new webpack.NoEmitOnErrorsPlugin(),
-    new ExtractTextPlugin({
-      filename: '[name].css',
-      disable: true,
```
Comment (erquhart) [inline webpack.dev.js]: Weird that this was even there, good catch. Wondering if we even need it at all for development, might be faster without. Not sure.

File: package.json [hunk 2]
```diff
@@ -126,6 +126,7 @@
   "dependencies": {
     "classnames": "^2.2.5",
     "create-react-class": "^15.6.0",
+    "express": "^4.16.2",
```
Comment (erquhart) [inline package.json]: This should move to devdependencies.

File: example/site/index.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
```
Comment (erquhart) [inline example/site/index.html]: I think this page is a great addition, just two adjustments I'd suggest:

1. Just style it without adding the material components library, not worth the addition
2. It should show on first load, but probably not every time

Thoughts?

File: example/site/index.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  <head>
+    <title>NetlifyCMS | Example</title>
+    <link rel="apple-touch-icon" sizes="180x180" href="https://res.cloudinary.com/netlify/image/upload/w_auto,c_scale/global/favicon/apple-touch-icon.png" />
+    <link rel="icon" type="image/png" href="https://res.cloudinary.com/netlify/image/upload/w_auto,c_scale/global/favicon/favicon-32x32.png" sizes="32x32" />
+    <link rel="icon" type="image/png" href="https://res.cloudinary.com/netlify/image/upload/w_auto,c_scale/global/favicon/favicon-16x16.png" sizes="16x16" />
+    <link rel="manifest" href="https://res.cloudinary.com/netlify/image/upload/w_auto,c_scale/global/favicon/manifest.json" />
+    <link rel="mask-icon" href="https://res.cloudinary.com/netlify/image/upload/w_auto,c_scale/global/favicon/safari-pinned-tab.svg" color="#00c7b7" />
+    <link rel="stylesheet" href="static/css/material-components-web.min.css">
+  </head>
+  <body>
```
Comment (erquhart) [inline example/site/index.html]: Nitpick: the splash content should be centered on the page.

File: example/site/index.html (hunk 3 above)
Comment (talves) [reply example/site/index.html]: > Just style it without adding the material components library, not worth the addition

It is just a development splash page. No use managing styling on the page. Any library can be used if you want to use a Netlify compiled style.

> It should show on first load, but probably not every time

Don't understand what you mean by this? Every time?

**Comment by erquhart** [comment]
@talves is this still relevant in light of your more recent work on the file system backend?

**Comment by talves** [comment]
Yeah, I will close this now. A restructure of dev is not as imperative with the file system backend as an option.

**Pull request closed without merging**